packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -781,16 +791,10 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
       // then skipping every frame and finishing in the new time.
       resetEpoch();
       _warmUpFrame = false;
+      warmUpFrameCompleter.complete();
       if (hadScheduledFrame)
         scheduleFrame();
     });
-
-    // Lock events so touch events etc don't insert themselves until the
-    // scheduled frame has finished.
-    lockEvents(() async {
-      await endOfFrame;
-      Timeline.finishSync();
-    });
```

Comment:
as far as i can tell, this patch shouldn't change the behaviour at all. endOfFrame is completed pretty much at the same time as handleDrawFrame ends, if I remember correctly.

Reply:
Actually, what would change is that we wouldn't unnecessarily schedule a frame, as `endOfFrame` is a [non-trivial getter](https://github.com/flutter/flutter/blob/16ff0b0da462175d5fd61c94d5134a6c131e2133/packages/flutter/lib/src/scheduler/binding.dart#L611-L623), that does schedule a frame. But for the bug it would not have any effect.